addons/stock/tests/test_inventory.py

```diff
@@ -305,13 +305,25 @@ def test_inventory_counted_quantity(self):
             'location_id': self.stock_location.id,
             'inventory_quantity': 42,
         })
-        # Generate new inventory, its line must have a theoretical
-        # quantity of 42 and a counted quantity to 0.
+        # Applies the change, the quant must have a quantity of 42 and a inventory quantity to 0.
         inventory_quant.action_apply_inventory()
         self.assertEqual(len(inventory_quant), 1)
         self.assertEqual(inventory_quant.inventory_quantity, 0)
         self.assertEqual(inventory_quant.quantity, 42)
 
+        # Checks we can write on `inventory_quantity_set` even if we write on
+        # `inventory_quantity` at the same time.
+        self.assertEqual(inventory_quant.inventory_quantity_set, False)
+        inventory_quant.write({'inventory_quantity': 5})
+        inventory_quant.flush()
```

Comment:
no need to flush, inventory_quantity_set depends of inventory_quantity then inventory_quantity_set is marked as dirty.